src/main/java/seedu/toluist/controller/FindController.java

```diff
@@ -0,0 +1,163 @@
+package seedu.toluist.controller;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.logging.Logger;
+
+import seedu.toluist.commons.core.LogsCenter;
+import seedu.toluist.dispatcher.CommandResult;
+import seedu.toluist.model.Task;
+import seedu.toluist.model.TodoList;
+import seedu.toluist.ui.Ui;
+
+/**
+ * Searches the task list for matches in the parameters, and displays the results received
+ */
+public class FindController extends Controller {
+    private static final String COMMAND_TEMPLATE = "^(?<command>(find|filter|list))"
+            + "(\\s+(?<keywords>.+))?\\s*";
+    private static final String SEARCH_BY_TAG = "tag/";
+    private static final String SEARCH_BY_NAME = "name/";
+    private static final String COMMAND_RESULT_TEMPLATE = "Searching for \"%s\" by %s.\n%d results found";
+    private static final String TRUE_COMMAND = "true";
+    private static final String FALSE_COMMAND = "false";
+    private static final String NAME_PARAMETER = "name";
+    private static final String TAG_PARAMETER = "tag";
+    private static final String NAME_AND_TAG_PARAMETER = "name and tag";
+    private static final String KEYWORDS = "keywords";
+    private static final String SPLITTER_COMMAND = " ";
+      
+    private final Logger logger = LogsCenter.getLogger(getClass());
+
+    public FindController(Ui renderer) {
+        super(renderer);
+    }
+
+    public CommandResult execute(String command) {
+        logger.info(getClass() + "will handle command");
+        
+        //initialize variables for searching
+        ArrayList<Task> foundTasks = new ArrayList<Task>();
+        HashMap<String, String> tokens = tokenize(command);
+        boolean isSearchByTag = tokens.get(SEARCH_BY_TAG).equals(TRUE_COMMAND);
+        boolean isSearchByName = tokens.get(SEARCH_BY_NAME).equals(TRUE_COMMAND);
+        String keywords = tokens.get(KEYWORDS);
+        String[] keywordList = keywords.split(" ");
+        keywordList = removeEmptySpaces(keywordList);
+        boolean isFound = false;
+        TodoList todoList = TodoList.load();
+        ArrayList<Task> taskList = todoList.getTasks();
+        Task currentTask;
+        int foundValue = 0;
+
+        for (int i = 0; i < taskList.size(); i++) {
```

Comment:
I would push these logic into the model actually, to make it more reusable. Controller's job is mainly to bring the data from the Model to the UI